test-cases/scylla-operator/longevity-scylla-operator-3h-terminate-replace.yaml

```diff
@@ -7,7 +7,11 @@ k8s_n_scylla_pods_per_cluster: 3
 n_loaders: 2
 n_monitor_nodes: 1
 
-nemesis_class_name: 'OperatorNodeTerminateAndReplace'
+# TODO: use 'TerminateKubernetesHostThenReplaceScyllaNode' when
```

Comment:
If we have 3 different nemesis should we have 3 different cases ? Or one nemesis set that would run them all ?

Reply:
The very first patch-set to this PR did exactly as you propose in the last statement. 
It was making unit tests fail.

Why?
Because creation of a nemesis with 3 methods causes `SisyphusMonkey` one ignore it's methods for itself when a nemesis filter is provided.

The cause for it is very strange and obfuscated logic for the nemesis methods handling.

Just imagine: SisyphusMonkey class, when is provided with nemesis filter, parses `string` variants of other nemesis classes definitions and looks for the "self.disrupt_foo(" stuff. Find `first` line and returns. It will pass if I have `comment` with such a string in class definition. Then, it returns only first occurrence. All other are ignored.

Reply:
There a list of classes that should be ignored in that scan.

Reply:
https://github.com/scylladb/scylla-cluster-tests/blob/1af3cc0f45c0d508b778376c260872fb39f89038/sdcm/nemesis.py#L4876

Reply:
> There a list of classes that should be ignored in that scan.

If it is ignored that these methods won't be picked up by the `SisyphusMonkey` complex nemesis. But I would expect it to be picked up.
The current logic, without changes, doesn't solve the current need.

Reply:
Ok, there is a way. Create A simple nemesis class for each of the methods and 2 complex ones. In that case we will be able to get new methods in both cases - only them and as part of the `SisyphusMonkey` one.

But the problem with it is that we add redundant code.